src/osparc_control/core.py

```diff
@@ -0,0 +1,378 @@
+from collections import deque
+from queue import Empty
+from queue import Queue
+from threading import Thread
+from time import sleep
+from typing import Any
+from typing import Deque
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Tuple
+from typing import Union
+from uuid import getnode
+from uuid import uuid4
+
+from pydantic import ValidationError
+from tenacity import Retrying
+from tenacity.stop import stop_after_delay
+from tenacity.wait import wait_fixed
+
+from .errors import CommnadNotAcceptedError
+from .errors import NoCommandReceivedArrivedError
+from .errors import NoReplyError
+from .models import CommandManifest
+from .models import CommandReceived
+from .models import CommandReply
+from .models import CommandRequest
+from .models import CommnadType
+from .models import RequestsTracker
+from .models import TrackedRequest
+from .transport.base_transport import SenderReceiverPair
+from osparc_control.transport.zeromq import ZeroMQTransport
+
+
+_MINUTE: float = 60.0
+
+WAIT_FOR_MESSAGES: float = 0.01
+WAIT_BETWEEN_CHECKS: float = 0.1
+# NOTE: this effectively limits the time between when
+# the two remote sides can start to communicate
+WAIT_FOR_RECEIVED: float = 1 * _MINUTE
+
+DEFAULT_LISTEN_PORT: int = 7426
+
+
+def _generate_request_id() -> str:
+    unique_hardware_id: int = getnode()
```

Comment:
How do you handle multiple control interface? they will have the same hardware id right?

Reply:
Multiple interfaces running on the same hardware will have the same id. I also added a `SESSION_ID` which is unique between runs. This should be more than enough to guarantee requests uniqueness.